dashboard/src/app/(authenticated)/userTaskDef/[...props]/components/UserTaskDef.tsx

```diff
@@ -1,18 +1,214 @@
-import { UserTaskDef as UserTaskDefProto } from 'littlehorse-client/dist/proto/user_tasks'
-import React, { FC } from 'react'
+'use client'
+
+import { UserTaskDef as UserTaskDefProto, UserTaskRunStatus } from 'littlehorse-client/dist/proto/user_tasks'
+import React, { FC, Fragment, useState } from 'react'
 import { Details } from './Details'
 import { Navigation } from '@/app/(authenticated)/components/Navigation'
 import { Fields } from './Fields'
+import { Button, Field, Input, Label } from '@headlessui/react'
+import { SEARCH_DEFAULT_LIMIT } from '@/app/constants'
+import Link from 'next/link'
+import { useInfiniteQuery } from '@tanstack/react-query'
+import { useWhoAmI } from '@/contexts/WhoAmIContext'
+import {
+  PaginatedUserTaskRunList,
+  searchUserTaskRun,
+} from '@/app/(authenticated)/userTaskDef/[...props]/actions/searchUserTaskRun'
+import { ArrowPathIcon } from '@heroicons/react/16/solid'
+import { concatWfRunIds, localDateTimeToUTCIsoString, utcToLocalDateTime } from '@/app/utils'
+import { useDebounce } from 'use-debounce'
+import { SearchFooter } from '@/app/(authenticated)/components/SearchFooter'
 
 type Props = {
   spec: UserTaskDefProto
 }
 export const UserTaskDef: FC<Props> = ({ spec }) => {
+  const userTaskPossibleStatuses = [
+    UserTaskRunStatus.ASSIGNED,
+    UserTaskRunStatus.CANCELLED,
+    UserTaskRunStatus.DONE,
+    UserTaskRunStatus.UNASSIGNED,
+  ]
+  const [selectedStatus, setSelectedStatus] = useState(UserTaskRunStatus.UNASSIGNED)
+  const [userId, setUserId] = useState('')
+  const [userGroup, setUserGroup] = useState('')
+  const [createdAfter, setCreatedAfter] = useState('')
+  const [createdBefore, setCreatedBefore] = useState('')
+  const { tenantId } = useWhoAmI()
+  const [limit, setLimit] = useState<number>(SEARCH_DEFAULT_LIMIT)
+  const [userIdToSearchFor] = useDebounce(userId, 1000)
+  const [userGroupToSearchFor] = useDebounce(userGroup, 1000)
```

Comment:
Why is useDebounce used here, what's the effect that it prevents if it's just a second?

Also `1000` is the delay, can it be a constant instead?

Reply:
it prevents to make server calls after every letter that the user types on the search boxes, with the debounce it will wait 1 second after the last typed character to call the server.
Yep I can extract that to a constant :)